DataFormats/FWLite/src/DataGetterHelper.cc

```diff
@@ -131,20 +133,22 @@ namespace fwlite {
         //obj.destruct();
         ////END OF WORK AROUND
 
-        TTreeCache* tcache = dynamic_cast<TTreeCache*> (branchMap_->getFile()->GetCacheRead());
+        if (tcUse_) {
+            TTreeCache* tcache = dynamic_cast<TTreeCache*> (branchMap_->getFile()->GetCacheRead());
 
-        if (nullptr == tcache) {
-            iData.branch_->GetEntry(eventEntry);
-        } else {
-            if (!tcTrained_) {
-                tcache->SetLearnEntries(100);
-                tcache->SetEntryRange(0, tree_->GetEntries());
-                tcTrained_ = true;
+            if (nullptr != tcache) {
+                if (!tcTrained_) {
+                    tcache->SetLearnEntries(100);
+                    tcache->SetEntryRange(0, tree_->GetEntries());
+                    tcTrained_ = true;
+                }
+                tree_->LoadTree(eventEntry);
             }
-            tree_->LoadTree(eventEntry);
-            iData.branch_->GetEntry(eventEntry);
-       }
-       iData.lastProduct_=eventEntry;
+        }
+        branchAccFoo_(iData.branch_);
```

Comment:
But DataGetterHelper doesn't have a ptr to the Event.

Note that I also had to change logick in DataGetterHelper::getBranchData() to honour initial request to not use an internal TTreeCache ... so that I can configure and control it externally.